eul/Views/Preference/PreferenceDisplayView.swift

```diff
@@ -49,6 +49,28 @@ extension Preference {
                     }
                 }
                 .frame(width: 200)
+                if #available(OSX 11.0, *) {
+                    Picker("appearance.mode".localized(), selection: $preference.appearanceMode) {
+                        ForEach(appearanceMode) {
+                            Text($0.description)
+                                .tag($0)
+                        }
+                    }
+                    .frame(width: 200)
+                    .onChange(of: preference.appearanceMode, perform: { _ in
+                        preference.changeColorScheme()
+
+                    })
+
+                } else {
+                    Picker("appearance.mode".localized(), selection: $preference.appearanceMode) {
+                        ForEach(appearanceMode) {
+                            Text($0.description)
+                                .tag($0)
+                        }
+                    }
+                    .frame(width: 200)
+                }
```

Comment:
what's the difference if we don't put `onChange` in the end?

Reply:
@gao-sun Do you know a way to respond to changes of a `Published` variable in the view model? I could not figure it out so I had to use `onChange`. The `onChange` detects if `appearanceMode` has changed, and calls the function to alter dark and light mode.

Reply:
i see. you can use `didSet` for listening changes of `@Published` in view model